src/include/binder/binder_context.h

```diff
@@ -37,10 +38,12 @@ class BinderContext {
   BinderContext() { upper_context = nullptr; }
 
   // Update the table alias map given a table reference (in the from clause)
-  void AddTable(const parser::TableRef* table_ref);
+  void AddTable(const parser::TableRef* table_ref,
+                concurrency::Transaction* txn = nullptr);
```

Comment:
I am not sure whether it is a good practice to set the default value as `nullptr`. Developers who are not familiar with the transaction logic may simply ignore passing the `txn` parameter, which subsequentally leads to the transaction bugs we found previously.